scripts/gulp/frontend-shared.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const cp = require('child_process');
+const fs = require('fs');
+const gulp = require('gulp');
+const babel = require('gulp-babel');
+const sourcemaps = require('gulp-sourcemaps');
+
+const buildFrontendSharedJs = () => {
+  // There does not appear to be a simple way of forcing gulp-babel to use a config
+  // file. Load it up as JSON and pass it in manually.
+  const babelConfig = JSON.parse(
+    fs.readFileSync('./frontend-shared/babel.config.json')
+  );
+
+  return (
+    gulp
+      .src('frontend-shared/src/*')
+      // Transpile the js source files and write the output in the frontend-shared/lib dir.
+      // Additionally, add the sourcemaps into the same dir.
+      .pipe(sourcemaps.init())
+      .pipe(babel(babelConfig))
+      .pipe(sourcemaps.write('.'))
+      .pipe(gulp.dest('frontend-shared/lib'))
```

Comment:
One major missing piece here. This does not delete the /lib dir first and we will want to do that for sure.

Reply:
I don't think it matters too much since consumers of the package are pulling specific files and shouldn't rely on what happens to be in `lib/`. When the package is built in CI, the directory will initially be empty.

Reply:
I did file a ticket to follow up here because its outputting sourcemaps for a production release which I think does not make sense.